spec/rspec/expectations/extensions/kernel_spec.rb

```diff
@@ -43,6 +43,16 @@ def method_missing(name, *args)
     it 'works properly on BasicObject-subclassed proxy objects' do
       expect(proxy_class.new(Object.new)).to be_proxied
     end
+
+    it 'does not break the deprecation check on BasicObject-subclassed proxy objects' do
+      begin
+        should_enabled = RSpec::Expectations::Syntax.should_enabled?
+        RSpec::Expectations::Syntax.enable_should unless should_enabled
+        proxy_class.new(BasicObject.new).should be_proxied
+      ensure
+        RSpec::Expectations::Syntax.disable_should if should_enabled
```

Comment:
@JonRowe Shouldn't we only disable though if it was previously disabled before? Without this check if it's enabled before this test starts it will be disabled after, which seems wrong.